third_party/move/move-binary-format/src/file_format.rs

```diff
@@ -1554,6 +1559,30 @@ impl SignatureToken {
             TypeParameter(idx) => subst_mapping[*idx as usize].clone(),
         }
     }
+
+    /// Can [self] be used in contexts where [other] is expected?
+    /// TODO(LAMBDA): extend beyond very simple function subtyping.
+    pub fn is_subtype_of(&self, other: &Self) -> bool {
+        use SignatureToken::*;
+        if self == other {
+            true
+        } else {
+            match (self, other) {
+                (Function(args, res, abilities), Function(args2, res2, abilities2)) => {
+                    // We need:
+                    // - Other has fewer abilities
+                    // - other's args are a subtypes of self's corresponding args
+                    // - self's results are subtypes of self's results
```

Comment:
Comment is incorrect (in that it is not what was intended, because the statement is tautological) as it makes no reference to `other`.